docs/development/development.md

```diff
@@ -0,0 +1,155 @@
+This document helps you get started using the Volcano code base.
+If you follow this guide and find some problem, please take
+a few minutes to update this file.
+
+- [Building the code](#building-the-code)
+- [Building docker images](#building-docker-images)
+- [Building a specific docker image](#building-a-specific-docker-image)
+- [Building the Volcano manifests](#building-the-volcano-manifests)
+- [Cleaning outputs](#cleaning-outputs)
+- [Running tests](#running-tests)
+- [Auto-formatting source code](#auto-formatting-source-code)
+- [Running the verification](#running-the-verification)
+- [Adding dependencies](#adding-dependencies)
+- [About testing](#about-testing)
+
+
+## Cloning the code
+
+You will need to clone the main `volcano` repo to `$GOPATH/src/volcano.sh/volcano` for
+the below commands to work correctly.
+
+## Building the code
+
+To build volcano all components for your host architecture, go to
+the source root and run:
+
+```bash
+make image_bins
+```
+
+To build a specific component for your host architecture, go to
+the source root and run `make <component name>`:
+
+```bash
+make vc-scheduler
+```
+
+
+## Building docker images
+
+Build the containers in your local docker cache:
+
+```bash
+make images
+```
+
+## Building a specific docker image
+
+If you want to make a local change and test some component, say `vc-controllers`, you
+could do:
+
+Under volcano.sh/volcano repo
+
+```bash
+pwd
+```
+The path should be
+
+```bash
+.../src/volcano.sh/volcano
+```
+
+Set up environment variables HUB and TAG by
+```bash
+export HUB=docker.io/yourrepo
+export TAG=citadel
+```
+
+Make some local change of the code, then build `vc-controllers`
+
+```bash
+make image.vc-controllers
+```
+
+## Building the Volcano manifests
+
+Use the following command to build the deploy yaml files:
+
+```bash
+make generate-yaml
+```
+
+## Cleaning outputs
+
+You can delete any build artifacts with:
+
+```bash
+make clean
+```
+
+## Running tests
+
+### Running unit tests
+
+You can run all the available unit tests with:
+
+```bash
+make unit-test
+```
+
+### Running e2e tests
+
+You can run all the available e2e tests with:
+
+```bash
+make vcctl
+make images
+make e2e-test-kind
+```
+
+If you want to run e2e test in a existing cluster with volcano deployed, run the following:
+
+```bash
+KUBECONFIG=${KUBECONFIG} go test ./test/e2e
```

Comment:
Need set the VK_BIN environment before running this go test command.

Reply:
ping @SrinivasChilveri

Reply:
Fixed